bika/lims/browser/bika_listing.py

```diff
@@ -594,7 +594,7 @@ def _process_request(self):
                     continue
                 ##logger.info("Or: %s=%s"%(index, value))
                 if idx.meta_type in('ZCTextIndex', 'FieldIndex'):
-                    self.Or.append(MatchRegexp(index, value))
+                    self.Or.append(MatchRegexp(index, '%s*' % value))
```

Comment:
Hmm that sounds surprising. A regexp like `term*` would match any string that contains `ter` with or without `m`, anywhere in the string (for `re.search()`) or at the start (for `re.match()`). That doesn't sound like what was intended. 

A glob search for `term*` would match strings starting with `term`. (The regexp would also match this, which looks correct, but it would also match `terq`, which is wrong).

I had a quick look at `AdvancedQuery` but it doesn't import `re` or `glob`. I don't know where the regexp evaluation actually happens. 

Regarding numbers, that's on the level of the index implementation, below `AdvancedQuery`. Text indexes normally filter out things like numbers and stopwords. To nicely match chemical names, you'll need a custom index. This is probably much better done using `collective.solr`. 

I see Bika doesn't use `ManagableIndex` ..

> ``AdvancedQuery`` works best when used together with
``Products.ManagableIndex`` and ``dm.incrementalsearch``.
Some of its features depend on these products, e.g. matching
and incremental filtering. Furthermore, these additional
components can speed up queries by several orders of magnitude.